activerecord/lib/active_record/associations/builder/has_one.rb

```diff
@@ -26,5 +32,31 @@ def self.define_validations(model, reflection)
         model.validates_presence_of reflection.name, message: :required
       end
     end
+
+    def self.touch_record(o, changes, name, touch, touch_method) # :nodoc:
+      record = o.send name
+
+      return unless record && record.persisted?
+
+      if touch
+        record.send(touch_method)
```

Comment:
`:touch` option can be given extra touched attribute name (e.g. `has_one :membership, touch: :joined_at`),
That is why `record.send(touch_method, touch)` if `touch != true` in belongs_to association.

https://github.com/rails/rails/blob/bc8697e1f642e40e511e5d5f0b459658d4b2724d/activerecord/lib/active_record/associations/builder/belongs_to.rb#L57-L61

Reply:
My bad. I have fixed this.